### ISSUE TITLE
DimPlot_scCustom error:Error in .subset2(x, i, exact = exact) : subscript out of bounds

### ISSUE THREAD

**Issue by Sophia409** [samuel-marsh/scCustomize#128]
Hi,

I can plot two metadata on umap by using funciton in Seurat, but couldn't plot it by using DimPlot_scCustom. Maybe you could help me figure out it here? 

Would be greatly appreciated! Best regards and many thanks!
Sophia

```
DimPlot(thalamus,pt.size = 1,label=T,label.size = 7,reduction = "umap",shuffle = T,raster = T,group.by = c('cell_type','development_stage') )
DimPlot_scCustom(Seuratob,pt.size = 1,label=T,label.size = 7,reduction = "umap",shuffle = T,raster = T,group.by = c('cell_type','development_stage'),num_columns = 2)
Error in .subset2(x, i, exact = exact) : subscript out of bounds
```

**Comment by samuel-marsh** [comment]
Hi Sophia,

Thanks for finding issue, I’ll see if I can figure out what issue is.  In meantime could you please post the output of `sessionInfo()` here so I know what versions are potential issues.

Thanks!
Sam

**Comment by Sophia409** [comment]
Hi Sam,

 This is my sessioninfo.

```
> sessionInfo()
R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22000)

Matrix products: default


locale:
[1] LC_COLLATE=Chinese (Simplified)_China.utf8  LC_CTYPE=Chinese (Simplified)_China.utf8    LC_MONETARY=Chinese (Simplified)_China.utf8
[4] LC_NUMERIC=C                                LC_TIME=Chinese (Simplified)_China.utf8    

time zone: Asia/Shanghai
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] scCustomize_1.1.3    org.Mm.eg.db_3.17.0  hgu95av2.db_3.13.0   org.Mmu.eg.db_3.17.0 devtools_2.4.5       usethis_2.2.2        datapasta_3.1.0     
 [8] cowplot_1.1.1        ggplot2_3.4.3        org.Hs.eg.db_3.17.0  AnnotationDbi_1.62.2 IRanges_2.34.1       S4Vectors_0.38.1     Biobase_2.60.0      
[15] BiocGenerics_0.46.0  dplyr_1.1.3          patchwork_1.1.3      SeuratObject_4.1.3   Seurat_4.3.0.1      

loaded via a namespace (and not attached):
  [1] RcppAnnoy_0.0.21        splines_4.3.0           later_1.3.1             prismatic_1.1.1         bitops_1.0-7            tibble_3.2.1           
  [7] polyclip_1.10-4         janitor_2.2.0           lifecycle_1.0.3         globals_0.16.2          processx_3.8.2          lattice_0.21-8         
 [13] MASS_7.3-60             magrittr_2.0.3          plotly_4.10.2           remotes_2.4.2.1         httpuv_1.6.11           sctransform_0.4.0      
 [19] sp_2.0-0                sessioninfo_1.2.2       pkgbuild_1.4.2          spatstat.sparse_3.0-2   reticulate_1.32.0       pbapply_1.7-2          
 [25] DBI_1.1.3               RColorBrewer_1.1-3      lubridate_1.9.2         abind_1.4-5             pkgload_1.3.2.1         zlibbioc_1.46.0        
 [31] Rtsne_0.16              purrr_1.0.1             RCurl_1.98-1.12         circlize_0.4.15         GenomeInfoDbData_1.2.10 ggrepel_0.9.3          
 [37] irlba_2.3.5.1           listenv_0.9.0           spatstat.utils_3.0-3    goftest_1.2-3           spatstat.random_3.1-6   fitdistrplus_1.1-11    
 [43] parallelly_1.36.0       leiden_0.4.3            codetools_0.2-19        shape_1.4.6             tidyselect_1.2.0        farver_2.1.1           
 [49] matrixStats_1.0.0       spatstat.explore_3.2-3  jsonlite_1.8.4          ellipsis_0.3.2          progressr_0.14.0        ggridges_0.5.4         
 [55] survival_3.5-5          tools_4.3.0             ica_1.0-3               Rcpp_1.0.10             glue_1.6.2              gridExtra_2.3          
 [61] xfun_0.40               GenomeInfoDb_1.36.3     withr_2.5.0             BiocManager_1.30.20     fastmap_1.1.1           fansi_1.0.4            
 [67] callr_3.7.3             digest_0.6.31           timechange_0.2.0        R6_2.5.1                mime_0.12               ggprism_1.0.4          
 [73] colorspace_2.1-0        scattermore_1.2         tensor_1.5              spatstat.data_3.0-1     RSQLite_2.3.1           utf8_1.2.3             
 [79] tidyr_1.3.0             generics_0.1.3          data.table_1.14.8       prettyunits_1.1.1       httr_1.4.6              htmlwidgets_1.6.2      
 [85] uwot_0.1.16             pkgconfig_2.0.3         gtable_0.3.4            blob_1.2.4              lmtest_0.9-40           XVector_0.40.0         
 [91] htmltools_0.5.5         profvis_0.3.8           scales_1.2.1            png_0.1-8               snakecase_0.11.1        knitr_1.44             
 [97] rstudioapi_0.14         tzdb_0.4.0              reshape2_1.4.4          nlme_3.1-162            curl_5.0.0              GlobalOptions_0.1.2    
[103] cachem_1.0.8            zoo_1.8-12              stringr_1.5.0           KernSmooth_2.23-21      vipor_0.4.5             parallel_4.3.0         
[109] miniUI_0.1.1.1          ggrastr_1.0.2           pillar_1.9.0            grid_4.3.0              vctrs_0.6.2             RANN_2.6.1             
[115] urlchecker_1.0.1        promises_1.2.0.1        xtable_1.8-4            cluster_2.1.4           paletteer_1.5.0         beeswarm_0.4.0         
[121] evaluate_0.21           readr_2.1.4             cli_3.6.1               compiler_4.3.0          rlang_1.1.1             crayon_1.5.2           
[127] future.apply_1.11.0     labeling_0.4.3          rematch2_2.1.2          ps_1.7.5                forcats_1.0.0           ggbeeswarm_0.7.2       
[133] plyr_1.8.8              fs_1.6.3                stringi_1.7.12          viridisLite_0.4.2       deldir_1.0-9            munsell_0.5.0          
[139] Biostrings_2.68.1       lazyeval_0.2.2          spatstat.geom_3.2-5     Matrix_1.5-4.1          hms_1.1.3               bit64_4.0.5            
[145] future_1.33.0           KEGGREST_1.40.0         shiny_1.7.4             highr_0.10              ROCR_1.0-11             igraph_1.5.1           
[151] memoise_2.0.1           bit_4.0.5              
```

**Comment by samuel-marsh** [comment]
Hi Sophia,

Found the issue which was due to a check I had for setting color palette.  I have fixed the issue in the "develop_V5" branch (v1.9.9.9014).  As SeuratV5 is backward compatible there should not be any issues updating onto this development branch.  However, if you due notice issues with this or any other functions upon updating to "develop_V5" while still using Seurat V4 please let me know!

Best,
Sam
